config.yaml

```diff
@@ -6,3 +6,11 @@ options:
      Enable GRUB cgroup overrides cgroup_enable=memory swapaccount=1. WARNING
      changing this option will reboot the host - use with caution on production
      services
+  http_proxy:
+     type: string
+     default:
+     description: HTTP proxy to embed as an environment setting for Docker
```

Comment:
The problem being with pulling it from the model ocnfig is if it changes you don't get the update in the charm. This is specifically to support HTTP proxy for prodstack style environments. Not all deployments need the proxy, and it needs to be reconfigurable on the job. 

I'm OK with sniffing from the env, but this brings in a new series of problems that we aren't yet ready to tackle. so I took the path of least resistance.